pkg/validator/container.go

```diff
@@ -176,39 +194,55 @@ func (cv *ContainerValidation) validateNetworking(networkConf *conf.Networking)
 func (cv *ContainerValidation) validateSecurity(securityConf *conf.Security) {
 	category := messages.CategorySecurity
 	securityContext := cv.Container.SecurityContext
+	podSecurityContext := cv.parentPodSpec.SecurityContext
+
+	// Support an empty container security context
 	if securityContext == nil {
 		securityContext = &corev1.SecurityContext{}
 	}
 
+	// Support an empty pod security context
+	if podSecurityContext == nil {
+		podSecurityContext = &corev1.PodSecurityContext{}
+	}
+
 	if securityConf.RunAsRootAllowed.IsActionable() {
-		if securityContext.RunAsNonRoot == (*bool)(nil) || !*securityContext.RunAsNonRoot {
-			cv.addFailure(messages.RunAsRootFailure, securityConf.RunAsRootAllowed, category)
-		} else {
+		if getBoolValue(securityContext.RunAsNonRoot) {
```

Comment:
Non-blocking: I think `getBoolValue(securityContext.RunAsNonRoot) || getBoolValue(podSecurityContext.RunAsNonRoot)` would simplify this a bit

Reply:
I found one more case where my code and this `||` implementation will return incorrect results. 

If the podspec is set to True, but the Container spec is explicitly False, then Polaris will not find a problem (when one exists). 

I have updated the code (pushing now)